liquidctl/driver/aura_led.py

```diff
@@ -0,0 +1,322 @@
+        0–255.
+        """
+        colors = iter(colors)
+        if _COLOR_MODES[mode].takes_color:
+            try:
+                r, g, b = next(colors)
+                single_color = (r, g, b)
+            except:
+                raise ValueError(f"one color required for this mode") from None
+        else:
+            single_color = (0, 0, 0)
+
+        if channel != "sync" and channel not in _COLOR_CHANNELS:
+            # _LOGGER.error("channel %s not valid", channel)
+            message = "valid channels are "
+            for chan in _COLOR_CHANNELS:
+                message += chan + " "
+            # _LOGGER.error(message)
+            raise KeyError(message) from None
+            return
+        """
+        This is experimental (it's an example of direct mode)
+        if mode == 'off':
+            self.channel_off(channel)
+            self.reset_all_channels()
+            return
+        """
+        if channel == "sync":
+            selected_channels = _COLOR_CHANNELS.values()
+        else:
+            selected_channels = (_COLOR_CHANNELS[channel],)
+        full_cmd_seq = []  # entire series of commands are added to this list
+        """
+        Experimental code for treating RGB channel differently from others
+        if channel == "rgb":
+            cmd_tuple=self.construct_color_commands(channel, mode, single_color)
+            self._write(cmd_tuple[0])
+            self._write(cmd_tuple[1])
+            self._write(_FUNCTION_CODE["end_seq2"])
+            self.end_color_sequence()
+            self._write(cmd_tuple[0])
+            self._write(cmd_tuple[1])
+            self._write(_FUNCTION_CODE["end_seq2"])
+        else:
+        """
+        for chan in selected_channels:
+            cmd_tuple = self.construct_color_commands(chan.name, mode, single_color)
+            full_cmd_seq.append(cmd_tuple[0])
+            full_cmd_seq.append(cmd_tuple[1])
+            full_cmd_seq.append(_FUNCTION_CODE["end_seq2"])
+            """
+            Asus Aura Crate sends command sequence twice, but our tests show
+            that this may be redundant. Nevertheless, let's keep this code here
+            in case we need to send commands twice as well
+            #full_cmd_seq.append(cmd_tuple[0])
+            #full_cmd_seq.append(cmd_tuple[1])
+            #full_cmd_seq.append(_FUNCTION_CODE["end_seq2"])
+            """
+        for cmd_seq in full_cmd_seq:
+            self._write(cmd_seq)
+        self.end_color_sequence()
+
+    def set_speed_profile(self, channel, profile, **kwargs):
+        """Not supported by this device."""
+        raise NotSupportedByDevice()
+
+    def set_fixed_speed(self, channel, duty, **kwargs):
+        """Not supported by this device."""
+        raise NotSupportedByDevice()
+
+    def reset_all_channels(self):
+        """Reset all LED channels."""
+        # for i in range(_RESET_CHANNEL_ITERATIONS):
+        for i in range(5):
```

Comment:
Asus Aura Crate software in Windows issues the 'reset' command 38 times in a row. This might be necessary in "direct" mode, but in "effect" mode it seems we can issue far fewer iterations. I'll change "_RESET_CHANNEL_ITERATIONS" to 5.